### PR TITLE
dev-java/lucene: 2.4.1-r5 build without BSFIX, drop java-ant-2 | java-utils-2.eclass: mark function use_doc as deprecated

### PR DESCRIPTION
**Pull request by vaukai**
---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by xgqt** [comment]
does the commit concerning abcl depends on those or any other PRs you made?

File: dev-lisp/abcl/abcl-1.9.2-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+BDEPEND=">=dev-java/ant-1.10.14-r3:0"
+DEPEND=">=virtual/jdk-1.8:*"
+RDEPEND=">=virtual/jre-1.8:*"
+
+src_compile() {
+	eant abcl.compile
+	eant abcl.jar || die "Can't make ABCL jar archive"
```
Comment (xgqt) [inline dev-lisp/abcl/abcl-1.9.2-r1.ebuild:23]: drop `|| die` ?

**Review by xgqt — CHANGES_REQUESTED** [review]

**Comment by vaukai** [comment]
> does the commit concerning abcl depends on those or any other PRs you made?

depends on https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=f1fce6210d9d4104738e3fa35214b477486be534 which is done.

File: dev-lisp/abcl/abcl-1.9.2-r1.ebuild (hunk 1 above)
Comment (vaukai) [reply dev-lisp/abcl/abcl-1.9.2-r1.ebuild]: > drop `|| die` ?

done.

**Review by xgqt — APPROVED** [review]

**Comment by vaukai** [comment]
@fordfrog all in this PR ready for review/merge.

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

**Comment by fordfrog** [comment]
on `lucene` i get some weird test failures:
```
    [junit] ------------- Standard Output ---------------
    [junit] Result: (fieldX:xxxxx fieldy:xxxxxxxx)^2.0
    [junit] ------------- ---------------- ---------------
    [junit] Testcase: testLegacyDateRange(org.apache.lucene.queryParser.TestQueryParser):	FAILED
    [junit] Query /[ 1.2.02 TO 4.2.02]/ yielded /[1.2.02 TO 4.2.02]/, expecting /[0cx572ow0 TO 0cxawu3jz]/
    [junit] junit.framework.AssertionFailedError: Query /[ 1.2.02 TO 4.2.02]/ yielded /[1.2.02 TO 4.2.02]/, expecting /[0cx572ow0 TO 0cxawu3jz]/
    [junit] 	at org.apache.lucene.queryParser.TestQueryParser.assertQueryEquals(TestQueryParser.java:144)
    [junit] 	at org.apache.lucene.queryParser.TestQueryParser.testLegacyDateRange(TestQueryParser.java:520)
    [junit] 
    [junit] 
    [junit] Testcase: testDateRange(org.apache.lucene.queryParser.TestQueryParser):	FAILED
    [junit] Query /default:[1.2.02 TO 4.2.02]/ yielded /[1.2.02 TO 4.2.02]/, expecting /[0cx572ow0 TO 0cxawu3jz]/
    [junit] junit.framework.AssertionFailedError: Query /default:[1.2.02 TO 4.2.02]/ yielded /[1.2.02 TO 4.2.02]/, expecting /[0cx572ow0 TO 0cxawu3jz]/
    [junit] 	at org.apache.lucene.queryParser.TestQueryParser.assertQueryEquals(TestQueryParser.java:154)
    [junit] 	at org.apache.lucene.queryParser.TestQueryParser.assertDateRangeQueryEquals(TestQueryParser.java:569)
    [junit] 	at org.apache.lucene.queryParser.TestQueryParser.testDateRange(TestQueryParser.java:538)
```
this is the related code:
```
  /** for testing legacy DateField support */
  private String getLegacyDate(String s) throws Exception {
    DateFormat df = DateFormat.getDateInstance(DateFormat.SHORT);
    return DateField.dateToString(df.parse(s));
  }
...
  /** for testing legacy DateField support */
  public void testLegacyDateRange() throws Exception {
    String startDate = getLocalizedDate(2002, 1, 1, false);
    String endDate = getLocalizedDate(2002, 1, 4, false);
    Calendar endDateExpected = Calendar.getInstance();
    endDateExpected.set(2002, 1, 4, 23, 59, 59);
    endDateExpected.set(Calendar.MILLISECOND, 999);
    assertQueryEquals("[ " + startDate + " TO " + endDate + "]", null,
                      "[" + getLegacyDate(startDate) + " TO " + DateField.dateToString(endDateExpected.getTime()) + "]");
    assertQueryEquals("{  " + startDate + "    " + endDate + "   }", null,
                      "{" + getLegacyDate(startDate) + " TO " + getLegacyDate(endDate) + "}");
  }
```
the expected result doesn't look like dates at all. i was ran using `openjdk-bin-8`.

running `-r4` brings the same results:
```
    [junit] ------------- Standard Output ---------------
    [junit] Result: (fieldX:xxxxx fieldy:xxxxxxxx)^2.0
    [junit] ------------- ---------------- ---------------
    [junit] Testcase: testLegacyDateRange(org.apache.lucene.queryParser.TestQueryParser):	FAILED
    [junit] Query /[ 1.2.02 TO 4.2.02]/ yielded /[1.2.02 TO 4.2.02]/, expecting /[0cx572ow0 TO 0cxawu3jz]/
    [junit] junit.framework.AssertionFailedError: Query /[ 1.2.02 TO 4.2.02]/ yielded /[1.2.02 TO 4.2.02]/, expecting /[0cx572ow0 TO 0cxawu3jz]/
    [junit] 	at org.apache.lucene.queryParser.TestQueryParser.assertQueryEquals(TestQueryParser.java:144)
    [junit] 	at org.apache.lucene.queryParser.TestQueryParser.testLegacyDateRange(TestQueryParser.java:520)
    [junit] 
    [junit] 
    [junit] Testcase: testDateRange(org.apache.lucene.queryParser.TestQueryParser):	FAILED
    [junit] Query /default:[1.2.02 TO 4.2.02]/ yielded /[1.2.02 TO 4.2.02]/, expecting /[0cx572ow0 TO 0cxawu3jz]/
    [junit] junit.framework.AssertionFailedError: Query /default:[1.2.02 TO 4.2.02]/ yielded /[1.2.02 TO 4.2.02]/, expecting /[0cx572ow0 TO 0cxawu3jz]/
    [junit] 	at org.apache.lucene.queryParser.TestQueryParser.assertQueryEquals(TestQueryParser.java:154)
    [junit] 	at org.apache.lucene.queryParser.TestQueryParser.assertDateRangeQueryEquals(TestQueryParser.java:569)
    [junit] 	at org.apache.lucene.queryParser.TestQueryParser.testDateRange(TestQueryParser.java:538)
```

when forcing `openjdk-8` instead, i still get the same failures. it would be best to debug into the code to see what's going on there but i can't do that now.

the rest of the prs i merged.

**Comment by fordfrog** [comment]
@vaukai did you do any changes or is it waiting for me to debug?

**Comment by vaukai** [comment]
> @vaukai did you do any changes or is it waiting for me to debug?

sorry for confusion. haven't changed anything here, only rebased and still waiting for you to debug.

**Comment by vaukai** [comment]
> @vaukai did you do any changes or is it waiting for me to debug?

@fordfrog once you find again time for debugging, i'd put prio 1 on https://github.com/gentoo/gentoo/pull/33582#issuecomment-2117872061, just for the sake of [java:21 stabilization](https://bugs.gentoo.org/showdependencytree.cgi?id=918647&hide_resolved=1).

**Comment by fordfrog** [comment]
@vaukai works for me now. i merged this pr. i think there should be a comment why we use `ant` instead of `eant` in the test so that nobody accidentally _fixes_ that...

**Comment by vaukai** [comment]
> @vaukai works for me now. i merged this pr. i think there should be a comment why we use `ant` instead of `eant` in the test so that nobody accidentally _fixes_ that...

i'll add such comment when doing next PR for removing the old version

**Pull request closed without merging**